infrastructure/smart-contracts/contracts/upgrades/esXai/esXai3.sol

```diff
@@ -0,0 +1,298 @@
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+import "../../Xai.sol";
+import "../../upgrades/referee/Referee7.sol";
+
+/**
+ * @title esXai
+ * @dev Implementation of the esXai
+ */
+contract esXai3 is ERC20Upgradeable, ERC20BurnableUpgradeable, AccessControlUpgradeable {
+
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+
+    EnumerableSetUpgradeable.AddressSet private _whitelist;
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    address public xai;
+    bool private _redemptionActive;
+    mapping(address => RedemptionRequest[]) private _redemptionRequests;
+    address public esXaiBurnFoundationRecipient;
+    uint256 public esXaiBurnFoundationBasePoints;
+    mapping(address => RedemptionRequestExt[]) private _extRedemptionRequests;
+    address public refereeAddress;
+    address public nodeLicenseAddress;
+    uint256 private maxKeysNonKyc; // TODO Get the initial value from management & confirm variable size needed
+
+    /**
+     * @dev This empty reserved space is put in place to allow future versions to add new
+     * variables without shifting down storage in the inheritance chain.
+     * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+     */
+    uint256[494] private __gap;
+
+    struct RedemptionRequest {
+        uint256 amount;
+        uint256 startTime;
+        uint256 duration;
+        bool completed;
+    }
+
+    struct RedemptionRequestExt {
+        uint256 amount;
+        uint256 startTime;
+        uint256 duration;
+        uint256 endTime;
+        bool completed;
+        bool cancelled;
+        uint256[5] __gap;
+    }
+
+    event WhitelistUpdated(address account, bool isAdded);
+    event RedemptionStarted(address indexed user, uint256 indexed index);
+    event RedemptionCancelled(address indexed user, uint256 indexed index);
+    event RedemptionCompleted(address indexed user, uint256 indexed index);
+    event RedemptionStatusChanged(bool isActive);
+    event XaiAddressChanged(address indexed newXaiAddress);
+    event FoundationBasepointsUpdated(uint256 newBasepoints);
+
+    function initialize (address _esXaiBurnFoundationRecipient, uint256 _esXaiBurnFoundationBasePoints, address _refereeAddress, address _nodeLicenseAddress, uint256 _maxKeys) public reinitializer(3) {
+        require(_esXaiBurnFoundationRecipient != address(0) && _esXaiBurnFoundationBasePoints <= 1000, "Invalid initialize");
+        require(_refereeAddress != address(0), "Invalid referee address");
+        require(_nodeLicenseAddress != address(0), "Invalid node license address");
+        esXaiBurnFoundationRecipient = _esXaiBurnFoundationRecipient;
+        esXaiBurnFoundationBasePoints = _esXaiBurnFoundationBasePoints;
```

Comment:
We should not have to initialize the esXaiBurnFoundationRecipient and esXaiBurnFoundationBasePoints again, they are already in the current proxy state

Reply:
This should be fixed now!